src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -3082,7 +3082,18 @@ private void UpdateDatabasesWithNewServerWideBackupConfiguration(TransactionOper
                     {
                         foreach (BlittableJsonReaderObject backup in backups)
                         {
-                            if (IsServerWideBackupWithTaskName(backup, periodicBackupConfiguration.Name))
+                            if (backup.TryGet(nameof(PeriodicBackupConfiguration.Name), out string backupName)  == false || backupName == null)
+                                throw new RachisInvalidOperationException(
+                                    $"This {nameof(PeriodicBackupConfiguration)} of {result.Key} has no value for {nameof(PeriodicBackupConfiguration.Name)}. This should not happens\n{backup}");
+                            
+                            if (serverWideBackupConfiguration.ToRemoveFromDatabaseRecord != null)
+                            {
+                                var toRemove = PutServerWideBackupConfigurationCommand.GetTaskNameForDatabase(serverWideBackupConfiguration.ToRemoveFromDatabaseRecord);
```

Comment:
You can move it up to avoid the `serverWideBackupConfiguration.ToRemoveFromDatabaseRecord != null` check